### DISCUSSION TITLE
Alternative way to create PDF Reports

### DISCUSSION THREAD

**Discussion by allwin199** [Cyfrin/security-and-auditing-full-course-s23#126] [General]
Vs code Extension: `Markdown PDF`

To create a page with a logo and titles

```js
<!DOCTYPE html>
<html>
<head>
<style>
    .full-page {
        width:  100%;
        height:  100vh; /* This will make the div take up the full viewport height */
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }
    .full-page img {
        max-width:  200;
        max-height:  200;
        margin-bottom: 5rem;
    }
    .full-page div{
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }
</style>
</head>
<body>

<div class="full-page">
    <img src="./logo.svg" alt="Logo">
    <div>
    <h1>Puppy Raffle Protocol Audit Report</h1>
    <h3>Prepared by: Prince Allwin</h3>
    </div>
</div>

</body>
</html>

<!-- Your report starts here! -->
.
.
.
```

[Example of a Report created using this extension](https://github.com/allwin199/puppy-raffle-audit/blob/main/audit-data/report.pdf)

[Example of a Markdown file](https://github.com/allwin199/puppy-raffle-audit/blob/main/audit-data/report.md)

@PatrickAlphaC Is there any downside to this method?

If there is no downside, we can add this to [audit-report-templating](https://github.com/Cyfrin/audit-report-templating) and link it in the [security-and-auditing-full-course-s23 Readme](https://github.com/Cyfrin/security-and-auditing-full-course-s23).

**Comment by truth2012** [comment]
You need to explain the procedure

**Reply by allwin199** [reply]
Step 1: Install the `Markdown PDF` extension in `vs code`
Step 2: [Get an example markdown file](https://github.com/Cyfrin/audit-report-templating/blob/main/report-example.md)
Step 3: Delete everything before `Table of contents` from the example markdown and replace it with [above example](https://github.com/Cyfrin/security-and-auditing-full-course-s23/discussions/126#discussion-6245253)
Step 3: Press `Cmd+Shift+P` If you are on Mac. `F1` on windows(I guess, not sure). Check the `Markdown PDF` extension for details.
Step 4: Type this command `markdown-pdf: Export (pdf)`

**Reply by truth2012** [reply]
Thank you I will try that
